### ISSUE TITLE
Tips

### ISSUE THREAD

**Issue by Torndeco** [EpochModTeam/EpochServer#1]
Just glanced over the code quickly, afew tips 

Use DLL_PROCESS_ATTACH  instead of doing an IF NULL check every time there is a callExtension.
https://github.com/EpochModTeam/EpochServer/blob/master/src/dllmain.cpp

Arma use of callExtension is a blocking call.
So we know that there will only be one thread at a time ever calling RVExtension.
Therefore you can optimize the callExtension by using allocating afew variables on the stack instead of allocating / deallocating on the heap all the time (in your callExtension function).
Basically stop allocating on the heap were you don't need to.

All the async threading is done by creating new threads & detaching them.
Creating Threads is overhead and costly, you should be re-using threads think worker thread pool.

There afew other things i saw, but the above are the main things.

**Comment by Torndeco** [comment]
Also i am curious, can someone explain why you are setting null character on Linux Builds ???
https://github.com/EpochModTeam/EpochServer/blob/master/src/main.cpp#L591-L594

**Comment by vbawol** [comment]
Thanks for the tips @Torndeco. 

@denisio @Fank Any here feedback guys?

**Comment by Fank** [comment]
Sure this can be optimized.
I choosed the "if NULL" check, because it was easier to write.
But sure the process attach and detach is much better.
